src/user/mod.rs

```
@@ -336,6 +339,86 @@ pub fn show(req: &mut Request) -> CargoResult<Response> {
     Ok(req.json(&R{ user: user.encodable() }))
 }
 
+#[derive(Insertable, Queryable, Identifiable, Associations)]
+#[belongs_to(User)]
+#[primary_key(user_id, target_id)]
+#[table_name="favorite_users"]
+pub struct FavoriteUser {
+    user_id: i32,
+    target_id: i32,
+}
+
+
+fn favorite_target(req: &mut Request) -> CargoResult<FavoriteUser> {
+    let user = req.user()?;
+    let target_user_id: i32 = req.params()["user_id"].parse().expect("User ID not found");
```

Do you think this should return an error rather than using `expect`?